### PR TITLE
Use tint methods.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Use tint methods to set drawable colors.

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Stypox** [comment]
What is the advantage of using these methods instead of `PorterDuffColorFilter`?

**Comment by TacoTheDank** [comment]
Hmm, this doesn't seem necessary in light of the coming support removal for APIs < 21. Opinions?

**Comment by Isira-Seneviratne** [comment]
> > Opinions?
> 
> This depends on whether we drop support for API 19 in 0.24.0 (spoiler: YES!) or not. 
> 
> > What is the advantage of using these methods instead of PorterDuffColorFilter?
> 
> Answer is awaiting ;-)

The PorterDuff mode does not need to be specified when setting the tint (the default is SRC_IN).

File: app/src/main/java/org/schabi/newpipe/player/Player.java [hunk 1]
```diff
@@ -484,10 +485,19 @@ private void initViews(@NonNull final PlayerBinding playerBinding) {
         binding.resizeTextView
                 .setText(PlayerHelper.resizeTypeOf(context, binding.surfaceView.getResizeMode()));
 
-        binding.playbackSeekBar.getThumb()
-                .setColorFilter(new PorterDuffColorFilter(Color.RED, PorterDuff.Mode.SRC_IN));
-        binding.playbackSeekBar.getProgressDrawable()
-                .setColorFilter(new PorterDuffColorFilter(Color.RED, PorterDuff.Mode.MULTIPLY));
+        tintThumbRed();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/Player.java:489]: Please remove these checks, as now the minsdk is lollipop (21), then this PR is ready

**Review by Stypox — CHANGES_REQUESTED** [review]

**Comment by TacoTheDank** [comment]
Could you rebase the PR (on dev) and force-push to only have one commit? That will make it much easier to read :)

File: app/src/main/java/org/schabi/newpipe/player/ui/VideoPlayerUi.java [hunk 2]
```diff
@@ -173,8 +172,9 @@ private void initViews() {
         playbackSpeedPopupMenu = new PopupMenu(context, binding.playbackSpeed);
         captionPopupMenu = new PopupMenu(themeWrapper, binding.captionTextView);
 
-        binding.progressBarLoadingPanel.getIndeterminateDrawable()
-                .setColorFilter(new PorterDuffColorFilter(Color.WHITE, PorterDuff.Mode.MULTIPLY));
+        binding.progressBarLoadingPanel.setIndeterminateTintList(ColorStateList.valueOf(Color
+                .WHITE));
```
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/player/ui/VideoPlayerUi.java:176]:
```suggestion
        binding.progressBarLoadingPanel.setIndeterminateTintList(
                ColorStateList.valueOf(Color.WHITE));
```

**Review by Stypox — CHANGES_REQUESTED** [review]
It doesn't seem to work correctly: the top image is the color of the seekbar before this PR, the bottom one is from this PR.
image
image

**Comment by TacoTheDank** [comment]
Sorry for being annoying lol, but I'd like to request another rebase for readability :P

**Comment by Stypox** [comment]
Now it seems to work as it should but... why does commit 763f0bf7ab31f3e7badb621c71cd4c8e83ffc086 solve the problem? I mean, why does `setThumbTintList` do the correct thing while `setProgressTintList` doesn't, requiring you to use `getProgressDrawable().setTint`? I still have some doubts about this PR, since it doesn't seem to provide better code than before (I mean, what was wrong with specifying `PorterDuff` mode?) but seems to introduce possibility for bugs.

**Comment by Isira-Seneviratne** [comment]
I undid a couple of the changes.

**Comment by Stypox** [comment]
I'm sorry, I have to reject this pull request. The original code calls `setColorFilter`, while the proposed code calls (I delved into library code) `setTintList` on the same `Drawable`. The two functions behave similarly, according to the [documentation](https://developer.android.com/reference/android/graphics/drawable/Drawable#setTintList(android.content.res.ColorStateList)), but there appear to be some differences, and I don't want to risk having non-working code. Also, [this Stack Overflow](https://stackoverflow.com/questions/56716093/setcolorfilter-is-deprecated-on-api29#56717316) thread suggests the differences actually are there.

**Pull request closed without merging**